### DISCUSSION TITLE
When I call pinMode(TFT_SCLK, OUTPUT) after tft.init(), the screen doesn't work

### DISCUSSION THREAD

**Discussion by baloyou** [Bodmer/TFT_eSPI#2797] [Q&A - General] [answered]
Hello,@Bodmer, thank you very much for your TFT_ESPI library, it helped me solve a lot of problems.

Now I have a tricky problem, I use the ESP32C3 microcontroller, and the ST7789-driven screen (1.3tft), which comes with a font chip (using the SPI communication protocol, sharing SCLK and MOSI pins with the screen)

This code doesn't work because I used it in loop 
```
  pinMode(TFT_SCLK, OUTPUT);
  pinMode(TFT_MOSI, OUTPUT);
```
It seems to be in conflict with the TFT_ESPI library, how should I fix it?

I added these two lines of code because I have another SPI device (font chip) and screen that share SCLK and MOSI pins.

Now my font library and screen can work separately, but putting them together will have problems.

Hope to get your help.

The following is a more complete code.

```
    #define TFT_MOSI 3
    #define TFT_SCLK 2
    #define TFT_CS   7  // Chip select control pin
    #define TFT_DC   6  // Data Command control pin
    #define TFT_RST   10

void io_init() {
  pinMode(TFT_SCLK, OUTPUT); 
  pinMode(TFT_MOSI, OUTPUT); 
}
void setup() {

  Serial.println("start");
  tft.init();
  tft.setSwapBytes(true);

  tft.fillScreen(TFT_BLACK);
  tft.setTextColor(TFT_WHITE, TFT_BLACK, true);
  tft.setFreeFont(&FreeSerifItalic24pt7b);
}

void loop() {

  io_init();

//  SET_FONT_STYLE (RED, BLACK, SONG_STYLE12); //先设置显示样式
//  DIS_CHINESE2(0, 120, DIS_CHINA);

  delay(1000);
  
  tft.fillScreen(TFT_BLACK);
  tft.drawChar(101, 50, 50);
  delay(1000);

  tft.fillScreen(TFT_BLACK);
  tft.drawChar(104, 50, 50);
  delay(1000);

}
```

**Comment by Bodmer** [comment, marked as answer]
To share the SPI pins with the TFT you will need to modify youur code to use the SPI class handle created by the TFT_eSPI library. You can then use that to configure the SPI parameters you need.

tft.getSPIinstance()

**Reply by baloyou** [reply]
Thank you very much, I think you are right.
